tagstudio/src/qt/flowlayout.py

```diff
@@ -23,101 +23,149 @@
 
 
 class FlowWidget(QWidget):
+    """
+    From what I understand:
+    A generic Qt window.
+    Please fix  my description
+    """
     def __init__(self, parent=None) -> None:
         super().__init__(parent)
         self.ignore_size: bool = False
 
 
 class FlowLayout(QLayout):
-    def __init__(self, parent=None):
+    """A generic Qt layout + the ability to keep track of the items in the layout."""
+    def __init__(self, parent=None) -> None:
+        # Initialize the parent class
         super().__init__(parent)
 
+        # In case parent is set
         if parent is not None:
+            # Set margins to 0.
             self.setContentsMargins(QMargins(0, 0, 0, 0))
 
-        self._item_list = []
-        self.grid_efficiency = False
+        # Initialize the items list to nothing.
+        self._item_list: list = []
+        # Initialize grid efficiency to "False"
+        self.grid_efficiency: bool = False
 
     def __del__(self):
         item = self.takeAt(0)
         while item:
             item = self.takeAt(0)
 
-    def addItem(self, item):
+    def addItem(self, item) -> None:
+        """Add a given item to the list of items inside the layout."""
         self._item_list.append(item)
 
-    def count(self):
+    def count(self) -> int:
+        """Returns the amount of items in the layout"""
         return len(self._item_list)
 
-    def itemAt(self, index):
-        if 0 <= index < len(self._item_list):
+    def itemAt(self, index: int):
+        """Returns the item at a given index or False if the index is out of range."""
+        # Try to get the value at index "index" (supports negative numbers).
+        try:
             return self._item_list[index]
+        except IndexError:
+            return False
 
-        return None
-
-    def takeAt(self, index):
-        if 0 <= index < len(self._item_list):
+    def takeAt(self, index: int):
+        """
+        Acts like the pop instructions of a list.
+        (Because that's exactly what it does but don't tell anyone.)
```

Comment:
👎

Reply:
Come on, it is funny

Reply:
yeah but this is not a place for standup comedy